src/ompl/base/OptimizationObjective.h

```diff
@@ -141,8 +144,12 @@ namespace ompl
             /** \brief Returns this objective's SpaceInformation. Needed for operators in MultiOptimizationObjective */
             const SpaceInformationPtr& getSpaceInformation() const;
 
+            /** \brief Allocate a heuristic-sampling state generator for this cost function, defaults to a basic rejection sampling scheme when the derived class does not provide a better method.*/
+            virtual InformedStateSamplerPtr allocInformedStateSampler(const StateSpace* space, const ProblemDefinitionPtr probDefn, const Cost* bestCost) const;
+
```

Comment:
I was mostly just trying to follow the existing pattern, obviously without really understanding the bigger design decisions...

I think a function like
```
InformedStateSamplerPtr SpaceInformation::informedSampler(const StateSpace* space, const OptimizationObjectivePtr opt)
```
would be fine? But such a function would still need to query the optimization objective, as ultimately the informed sampler is specific to the optimization objective...